### ISSUE TITLE
Value cannot be null. (Parameter 'provider')

### ISSUE THREAD

**Issue by thepont** [Azure/azure-functions-core-tools#1995]
Similar issue to #1924. When I type `func start` in a nodejs project on Arch Linux I am getting the following error error.

```
Hosting environment: Production
Content root path: /home/pont/projects/fibre-collective/packages/certification
Now listening on: http://0.0.0.0:7071
Application started. Press Ctrl+C to shut down.
Value cannot be null. (Parameter 'provider')
```
enabling `CLI_DEBUG` results in the following trace:

```
Azure Functions Core Tools (3.0.2358 Commit hash: d6d66f19ea30fda5fbfe068fc22bc126f0a74168)
Function Runtime Version: 3.0.13159.0
[5/19/2020 12:09:26 PM] FUNCTIONS_WORKER_RUNTIME set to node. Skipping WorkerConfig for language:python
[5/19/2020 12:09:26 PM] FUNCTIONS_WORKER_RUNTIME set to node. Skipping WorkerConfig for language:java
[5/19/2020 12:09:26 PM] FUNCTIONS_WORKER_RUNTIME set to node. Skipping WorkerConfig for language:powershell
[5/19/2020 12:09:26 PM] Building host: startup suppressed: 'False', configuration suppressed: 'False', startup operation id: '190e8a62-6b3e-48f1-83d1-a3ede0bce239'
[5/19/2020 12:09:26 PM] Reading host configuration file '/home/pont/projects/fibre-collective/packages/certification/host.json'
[5/19/2020 12:09:26 PM] Host configuration file read:
[5/19/2020 12:09:26 PM] {
[5/19/2020 12:09:26 PM]   "version": "2.0",
[5/19/2020 12:09:26 PM]   "logging": {
[5/19/2020 12:09:26 PM]     "applicationInsights": {
[5/19/2020 12:09:26 PM]       "samplingExcludedTypes": "Request",
[5/19/2020 12:09:26 PM]       "samplingSettings": {
[5/19/2020 12:09:26 PM]         "isEnabled": true
[5/19/2020 12:09:26 PM]       }
[5/19/2020 12:09:26 PM]     }
[5/19/2020 12:09:26 PM]   },
[5/19/2020 12:09:26 PM]   "extensionBundle": {
[5/19/2020 12:09:26 PM]     "id": "Microsoft.Azure.Functions.ExtensionBundle",
[5/19/2020 12:09:26 PM]     "version": "[1.*, 2.0.0)"
[5/19/2020 12:09:26 PM]   }
[5/19/2020 12:09:26 PM] }
[5/19/2020 12:09:26 PM] Reading functions metadata
[5/19/2020 12:09:26 PM] 1 functions found
[5/19/2020 12:09:26 PM] Looking for extension bundle Microsoft.Azure.Functions.ExtensionBundle at /tmp/Functions/ExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle
[5/19/2020 12:09:26 PM] Fetching information on versions of extension bundle Microsoft.Azure.Functions.ExtensionBundle available on https://functionscdn.azureedge.net/public/ExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/index.json
[5/19/2020 12:09:27 PM] Downloading extension bundle from https://functionscdn.azureedge.net/public/ExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/1.1.1/Microsoft.Azure.Functions.ExtensionBundle.1.1.1.zip to /tmp/d87a8cf9-a864-42f0-a891-b7e41a7d103d/Microsoft.Azure.Functions.ExtensionBundle.1.1.1.zip
Hosting environment: Production
Content root path: /home/pont/projects/fibre-collective/packages/certification
Now listening on: http://0.0.0.0:7071
Application started. Press Ctrl+C to shut down.
[5/19/2020 12:11:22 PM] Executing HTTP request: {
[5/19/2020 12:11:22 PM]   "requestId": "90571ec9-f3f7-4606-b012-62097466c9a3",
[5/19/2020 12:11:22 PM]   "method": "GET",
[5/19/2020 12:11:22 PM]   "uri": "/"
[5/19/2020 12:11:22 PM] }
System.ArgumentNullException: Value cannot be null. (Parameter 'provider')
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Azure.Functions.Cli.Actions.HostActions.StartHostAction.RunAsync() in D:\a\1\s\src\Azure.Functions.Cli\Actions\HostActions\StartHostAction.cs:line 252
   at Azure.Functions.Cli.ConsoleApp.RunAsync[T](String[] args, IContainer container) in D:\a\1\s\src\Azure.Functions.Cli\ConsoleApp.cs:line 66
Application is shutting down...
[5/19/2020 12:11:45 PM] Stopping host...
[5/19/2020 12:11:45 PM] Host shutdown completed.
```

I have attempted to find and delete the Extension bundles on my system haven't been able to.

This seems to be the corresponding line where it is quitting.
https://github.com/Azure/azure-functions-core-tools/blob/d8d237f44132541b857ab16314dff1e843025c11/src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs#L265

**Comment by anthonychu** [comment]
@soninaren This might be a bundles problem. Could you please take a look?

**Comment by soninaren** [comment]
> @soninaren This might be a bundles problem. Could you please take a look?

Sure, From the core tools output it does look like the host is starting before the bundle download is complete.

@thepont did this work on retry. Can you delete this directory `/tmp/d87a8cf9-a864-42f0-a891-b7e41a7d103d/`. Execute `func new` -> `Ctrl + c` and then do `func start`

**Comment by thepont** [comment]
@soninaren thanks for your help here, That has indeed solved the issue for me.

I assume the download was partial due to my internet connection being a bit flakie.

**Comment by anthonychu** [comment]
Glad you were able to fix the problem! I've created #1999 to make some improvements as this happens occasionally and it's not obvious how to fix it.
